Module 'struct' has no 'unpack' member (no-member)

### Bug description

```python
import struct
struct.unpack('h', b'\x00\x01')
```

### Configuration

_No response_

### Command used

```shell
v/bin/pylint -E 1.py
```

### Pylint output

```shell
************* Module 1
1.py:2:0: E1101: Module 'struct' has no 'unpack' member (no-member)
```


### Expected behavior

_No response_

### Pylint version

```shell
$ v/bin/pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.8.12 (default, Sep 21 2021, 00:10:52) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
```

### OS / Environment

CentOS 8 Stream

### Additional dependencies

```shell
$ v/bin/pip list
Package           Version
----------------- -------
astroid           2.9.3
isort             5.10.1
lazy-object-proxy 1.7.1
mccabe            0.6.1
pip               21.3.1
platformdirs      2.5.0
pylint            2.12.2
setuptools        60.5.0
toml              0.10.2
typing_extensions 4.0.1
wheel             0.37.1
wrapt             1.13.3
```

Facing the same issue with quite different versions:

```
% pylint --version
pylint 2.10.2
astroid 2.7.3
Python 3.9.9 (main, Nov 21 2021, 03:23:42) 
[Clang 13.0.0 (clang-1300.0.29.3)]
% cat silly.py 
import struct
print(struct.unpack('i', b'\01\00\00\00'))
% pylint -E silly.py
************* Module silly
silly.py:2:6: E1101: Module 'struct' has no 'unpack' member (no-member)
```

--- a/tests/functional/r/regression_02/regression_5801.py
+++ b/tests/functional/r/regression_02/regression_5801.py
@@ -1,4 +1,5 @@
-# pylint: disable=missing-module-docstring, missing-docstring
+# https://github.com/PyCQA/pylint/issues/5801
+# pylint: disable=missing-docstring
 
 import struct
 struct.unpack('h', b'\x00\x01')

--- a/tests/functional/r/regression_02/regression_5801.py
+++ b/tests/functional/r/regression_02/regression_5801.py
@@ -1,0 +1,4 @@
+# pylint: disable=missing-module-docstring, missing-docstring
+
+import struct
+struct.unpack('h', b'\x00\x01')

